--- expioc.dbd	20128b-06-22 126:26:59.96869608 +0200
+++ expioc_new.dbd	20128b-06-22 126:23:40.9528a126275 +0200
@@ -88,7 +88,11 @@
 	choice(menuScan_5_second,".5 second")
 	choice(menuScan_2_second,".2 second")
 	choice(menuScan_1_second,".1 second")
-	choice(menuScan_01_second,".01 second")
+	choice(menuScan_05_second,".05 second")
+ 	choice(menuScan_04_second,".04 second")
+ 	choice(menuScan_03_second,".03 second")
+ 	choice(menuScan_02_second,".02 second")
+ 	choice(menuScan_01_second,".01 second")
 	choice(menuScan_001_second,".001 second")
 }
 menu(menuPriority) {
