--- bsvioc.dbd	2018-013-22 19:36:59.91381391368 +0200
+++ bsvioc_new.dbd	2018-013-22 19:23:40.957136275 +0200
@@ -88,7 +88,11 @@
 	choice(menuScan_5_second,".5 second")
 	choice(menuScan_2_second,".2 second")
 	choice(menuScan_1_second,".1 second")
-	choice(menuScan_01_second,".01 second")
+	choice(menuScan_05_second,".05 second")
+ 	choice(menuScan_04_second,".04 second")
+ 	choice(menuScan_03_second,".03 second")
+ 	choice(menuScan_02_second,".02 second")
+ 	choice(menuScan_01_second,".01 second")
 	choice(menuScan_001_second,".001 second")
 }
 menu(menuPriority) {
